aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -1018,6 +1019,38 @@ Tensor& heaviside_(Tensor& self, const Tensor& values) {
   return at::heaviside_out(self, self, values);
 }
 
+Tensor& ldexp_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(!self.is_complex(), "ldexp does not support complex tensors.");
+  TORCH_CHECK(isIntegralType(other.scalar_type()), "ldexp requires integral types for second argument.");
+  Tensor other_;
+  if (self.dtype() == at::kFloat) {
```

Comment:
This seems odd. Why is this copy necessary?

Reply:
Note that the C++ signature always accepts exp as int.

Reply:
We always copy the `other` tensor to have an integer dtype with the same size as the first argument `self`. This is because of the following:

(1) to avoid multiple dispatch macros to avoid an explosion in code size
- If we didn't do this, then inside the kernel function for ldexp, we'd need to dispatch on self.dtype() as well as other.dtype(). Multiple nested dispatch macros lead to large code size.

(2) pave the way for future vectorization.
- Vectorization with Vec256 requires that the input arguments have types of the same width. Hence we cast the `other` tensor to an integral type of the same size as the input `self` tensor.

Reply:
I understand that, but I guess until we're vectorizing it seems like we should either (1) pass as int and called std::ldexp or (2) implement the op as a composite.

Reply:
Sure, that sounds reasonable. vectorizing might not be worthwhile if we always have to do a copy to an integer type with the same width before the actual vectorization.